File: src/com/libill/demos/activity/LifecycleActivity.java
package com.libill.demos.activity;

import java.text.SimpleDateFormat;
import java.util.Date;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.text.method.ScrollingMovementMethod;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import android.widget.TextView;

public class LifecycleActivity extends Activity {
	public String TAG = "LifecycleActivity";
	private TextView textView;
	SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd-HH:mm:ss SSS");

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		printlnLog("onCreate", getIntent());

		textView = new TextView(this);
		// 使TextView可以滚动
		textView.setMovementMethod(new ScrollingMovementMethod());
		setContentView(textView);
		textView.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				Intent intent = new Intent(LifecycleActivity.this, LifecycleActivity.class);
				intent.putExtra("aTestContent","This is a test content.");
				startActivity(intent);
			}
		});
		logAndSetText("onCreate");
	}


	@Override
	protected void onNewIntent(Intent intent) {
		super.onNewIntent(intent);
		printlnLog("onNewIntent", intent);
	}

	@Override
	protected void onStart() {
		super.onStart();
		logAndSetText("onStart");
	}

	@Override
	protected void onResume() {
		super.onResume();
		logAndSetText("onResume");
	}

	@Override
	public void onWindowFocusChanged(boolean hasFocus) {
		super.onWindowFocusChanged(hasFocus);
		if (hasFocus) {
			logAndSetText("onWindowFocusChanged");
		}
	}

	@Override
	protected void onPause() {
		super.onPause();
		logAndSetText("onPause");
	}

	@Override
	protected void onStop() {
		super.onStop();
		logAndSetText("onStop");
	}

	@Override
	protected void onDestroy() {
		super.onDestroy();
		logAndSetText("onDestroy");
	}

	@Override
	public boolean onTouchEvent(MotionEvent event) {
		Intent intent = new Intent(this, LifecycleActivity.class);
		intent.putExtra("aTestContent","This is a test content.");
		startActivity(intent);
		return super.onTouchEvent(event);
	}

	public void logAndSetText(String text) {
		Log.i(TAG, text);
		textView.setText(textView.getText() + "\n" + sdf.format(new Date()) + " : " + text);
	}

	public void printlnLog(String tag, Intent intent) {
		String content = intent.getStringExtra("aTestContent");
		Log.i(tag, content);
	}
}

File: app/src/main/java/com/libill/demos/activity/LifecycleActivity.kt
package com.libill.demos.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.text.method.ScrollingMovementMethod
import android.util.Log
import android.view.MotionEvent
import android.widget.TextView
import com.libill.test.R
import java.text.SimpleDateFormat
import java.util.Date

class LifecycleActivity : Activity() {
    companion object {
        private const val TAG: String = "LifecycleActivity"
    }

    private var textView: TextView? = null
    private var sdf: SimpleDateFormat = SimpleDateFormat("yyyy-MM-dd-HH:mm:ss SSS")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        printlnLog("onCreate", intent)

        textView = TextView(this)
        textView!!.setTextColor(resources.getColor(R.color.black))
        // 使TextView可以滚动
        textView!!.movementMethod = ScrollingMovementMethod()
        setContentView(textView)
        textView!!.setOnClickListener {
            val intent = Intent(
                this@LifecycleActivity,
                LifecycleActivity::class.java
            )
            intent.putExtra("aTestContent", "This is a test content.")
            startActivity(intent)
        }
        logAndSetText("onCreate")
    }


    override fun onNewIntent(intent: Intent) {
        super.onNewIntent(intent)
        printlnLog("onNewIntent", intent)
    }

    override fun onStart() {
        super.onStart()
        logAndSetText("onStart")
    }

    override fun onResume() {
        super.onResume()
        logAndSetText("onResume")
    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            logAndSetText("onWindowFocusChanged")
        }
    }

    override fun onPause() {
        super.onPause()
        logAndSetText("onPause")
    }

    override fun onStop() {
        super.onStop()
        logAndSetText("onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        logAndSetText("onDestroy")
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        val intent = Intent(this, LifecycleActivity::class.java)
        intent.putExtra("aTestContent", "This is a test content.")
        startActivity(intent)
        return super.onTouchEvent(event)
    }

    fun logAndSetText(text: String) {
        Log.i(TAG, text)
        textView!!.text = """${textView!!.text}${sdf.format(Date())} : $text"""
    }

    private fun printlnLog(tag: String?, intent: Intent) {
        val content = intent.getStringExtra("aTestContent")
        Log.i(tag, content!!)
    }
}
